ci/pipeline.yml

```diff
@@ -0,0 +1,305 @@
+        <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|View build details>
+
+- name: deploy-registry-production
+  serial_groups: [mirror]
+  plan:
+  - in_parallel: 
+    - get: registry-source
+      params: {depth: 1}
+      passed: [integration-test]
+    - get: docker-registry-image
+      passed: [integration-test]
+    - get: ci-source
+      params: {depth: 1}
+  - task: create-s3-bucket
+    file: ci-source/ci/tasks/create-s3-bucket.yml
+    params:
+      <<: *production-cf
+  - task: deploy-registry
+    file: ci-source/ci/tasks/deploy-registry.yml
+    params:
+      <<: *production-cf
+    on_failure:
+      try:
+        task: cancel-deployment
+        file: ci-source/ci/tasks/cancel-deployment.yml
+        params:
+          <<: *production-cf
+          CF_APP_NAME: docker-registry-mirror
+  on_failure:
+    put: slack
+    params: &slack-params
+      text: |
+        :x: FAILED to deploy docker registry mirror to production. Deployment was canceled (rolled back). 
+        <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|View build details>
+      channel: ((slack-channel))
+      username: ((slack-username))
+      icon_url: ((slack-icon-url))  
+
+- name: deploy-proxy-production
+  serial_groups: [proxy]
+  plan:
+  - in_parallel:
+    - get: nginx-conf
+      passed: [integration-test]
+    - get: proxy-source
+      params: {depth: 1}
+      passed: [integration-test]
+    - get: ci-source
+      params: {depth: 1}
+  - task: deploy-proxy
+    file: ci-source/ci/tasks/deploy-proxy.yml
+    tags: [iaas]
+    params: 
+      <<: *production-cf
+    on_failure:
+      try: 
+        task: cancel-deployment
+        file: ci-source/ci/tasks/cancel-deployment.yml
+        params:
+          <<: *production-cf
+          CF_APP_NAME: docker-registry-proxy
+  on_failure:
+    put: slack
+    params:
+      <<: *slack-params
+      text: |
+        :x: FAILED to deploy docker registry mirror proxy to production. Deployment was canceled (rolled back).
+        <$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|View build details>
+
+resources:
+
+- name: pipeline-config
+  type: s3-iam
+  tags: [iaas]
+  source:
+    bucket: ((concourse-creds-bucket))
+    region_name: ((concourse-creds-bucket-region))
+    versioned_file: docker-registry-mirror.yml
+
+- name: pipeline-source
+  type: git
+  source:
+    uri: https://github.com/cloud-gov/docker-registry-mirror.git
+    paths: [ci/pipeline.yml]
+
+- name: ci-source
+  type: git
+  source:
+    uri: https://github.com/cloud-gov/docker-registry-mirror.git
+    paths: [ci/*]
+
+- name: nginx-conf
+  type: s3-iam
+  tags: [iaas]
+  source:
+    bucket: ((concourse-varz-bucket))
+    region_name: ((concourse-varz-bucket-region))
+    versioned_file: docker-registry-mirror/nginx/nginx.conf
```

Comment:
The `nginx.conf` is stored in S3?  Do we have documentation on where/how to maintain this file elsewhere internally?

Reply:
It is where all the other config is. You can always fetch the pipeline and see values. I wasn't sure if I should mention that here in a public repo or not. Thoughts?

Reply:
That's a good question - it's referenced here in the code, hence my initial question. :-) I agree though, any details around this should live elsewhere as it's sensitive - @pburkholder, do you have any thoughts or recommendations one way or another?

Reply:
We'll tackle this separately if we feel we need to.  🚀